### PR TITLE
Multicollateral bonding curve pool initialization

### PR DESCRIPTION
**Pull request by bragov4ik**
* Add initialization for mcbc source akin to other sources
* Better test organization in `qa-tools`
* Remove redundant type parameter in `qa-tools` pallet (whitelist capacity; whitelist was removed previously)
* Small improvements of `qa-tools` (readability, code deduplication)

### CODE REVIEW

File: pallets/qa-tools/Cargo.toml [hunk 1]
```diff
@@ -37,6 +39,7 @@ framenode-runtime = { path = "../../runtime", default-features = false, features
 sp-core = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
 sp-io = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
 sp-runtime = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
+pallet-balances = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
```
Comment (wer1st) [inline pallets/qa-tools/Cargo.toml:42]: aplhabetical order

File: pallets/qa-tools/src/lib.rs [hunk 2]
```diff
@@ -362,6 +377,50 @@ pub mod pallet {
...
+            tbcd_collateral: Option<pallet_tools::mcbc::TbcdCollateralInput>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+            if let Some(base_supply) = base_supply {
+                pallet_tools::mcbc::initialize_base_supply::<T>(base_supply)?;
+            }
+
+            // handle tbcd collateral first as it may initialize reference asset xor prices
+            // (initialization of all collateral is dependant on these prices)
+            let mut collateral_ref_prices = vec![];
```
Comment (wer1st) [inline pallets/qa-tools/src/lib.rs:395]:
```suggestion
            let mut collateral_ref_prices = Vec::new();
```

Comment (wer1st) [reply pallets/qa-tools/src/lib.rs:395]: or BTreeMap

File: pallets/qa-tools/src/lib.rs [hunk 3]
```diff
@@ -362,6 +377,50 @@ pub mod pallet {
...
+                if let Some(actual_ref_prices) = actual_ref_prices {
+                    collateral_ref_prices.push((common::TBCD.into(), actual_ref_prices));
+                }
+            }
+            for collateral_input in other_collaterals {
+                let collateral_asset_id = collateral_input.asset;
+                let actual_ref_prices =
+                    pallet_tools::mcbc::initialize_single_collateral::<T>(collateral_input)?;
+                if let Some(actual_ref_prices) = actual_ref_prices {
+                    Self::deposit_event(Event::<T>::OrderBooksFilled);
```
Comment (bragov4ik) [inline pallets/qa-tools/src/lib.rs:408]: what.......)))

**Review by wer1st — APPROVED** [review]

**Pull request merged**